conans/paths.py

```diff
@@ -4,15 +4,43 @@
 import platform
 from conans.errors import ConanException
 from conans.util.files import rmdir
-
-if platform.system() == "Windows":
-    from conans.util.windows import path_shortener, rm_conandir, conan_expand_user
+import sys
+
+
+def _check_long_paths_support():
+    if platform.system() != "Windows":
+        return True
+    if sys.version_info < (3, 6):
+        return False
+    from six.moves import winreg
+    try:
+        hKey = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE,
+                              r"SYSTEM\CurrentControlSet\Control\FileSystem")
+        result = winreg.QueryValueEx(hKey, "LongPathsEnabled")
+        key_value = result[0]
+        return key_value == 1
```

Comment:
I would suggest actually trying to create a very long path and seeing that it works after reading the reg key. The reason is that you can have the reg key on without rebooting in which case it is not in effect.

Reply:
You sure? I have manually modified the reg key with regedit, and it seems it worked. Also seems a bit of a corner case to be running such test in every conan execution?

Reply:
You are right, I remembered this incorrectly:

> The registry key will not be reloaded during the lifetime of the process. In order for all apps on the system to recognize the value of the key, a reboot might be required because some processes may have started before the key was set.

https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx

To be fair, this still implies that there can be a situation where the key is on and has no effect. Pathological, but can save you an issue in the future :)